### PR TITLE
Dropdown list (Mobile): Min width

### PR DESCRIPTION
**Pull request by grzegorz-czajkowski**
Two commits:
 * UIComponents (Mobile): add Dropdown list (Min width)
 * Dropdown list (Mobile): apply alignment according to guide

### CODE REVIEW

File: src/js/profile/mobile/widget/DropdownMenu.js [hunk 1]
```diff
@@ -1010,17 +1011,19 @@
...
 				parseInt(stylesOfSelectedOptionAfter.width, 10) : wrapperMinWidth;
 
 				height = optionHeight;
+
 				// This part decides the location and direction of option list.
-				offsetLeft = self._offset.left;
+				offsetLeft = self._horizontalDirection === "right" ? window.screen.width - width : 0;
 				optionStyle = "left: " + offsetLeft + "px; ";
+
 				if (options.inline === true) {
```
Comment (DohyungLim) [inline src/js/profile/mobile/widget/DropdownMenu.js:1032]: Type is boolean, do I need "true"? If it can be omitted, nothing is concise.

**Review by DohyungLim — COMMENTED** [review]
1 comment need to check.

File: src/js/profile/mobile/widget/DropdownMenu.js (hunk 1 above)
Comment (grzegorz-czajkowski) [reply src/js/profile/mobile/widget/DropdownMenu.js:1032]: Are you refering to this line?
`if (options.inline === true)`
I am not changing this and could be refactored separately.

**Comment by grzegorz-czajkowski** [comment]
Solved most issues related to alignment position
 - added top and bottom margins if dropdown is only opened from non list item element (for example appabar)
 - dropdow list is shown of left or right most position based on click position
 - fixed alignment on list (avoid addition 12px from top, added from js code).
Created additional issue https://github.com/Samsung/TAU/issues/808 where we need to take into consideration list position to always show dropdown menu within list.

File: src/css/profile/mobile/common/dropdownmenu.less [hunk 2]
```diff
@@ -194,6 +191,11 @@
 		border-top: 1 * @px_base solid @color_dropdownmenu_list_group_stroke;
 		border-bottom: 1 * @px_base solid @color_dropdownmenu_list_group_stroke;
 	}
+	// top, bottom classes just describes direction in which menu is shown.
```
Comment (TomaszLukawskiSam) [inline src/css/profile/mobile/common/dropdownmenu.less]: you don't need define empty styles.

File: src/js/profile/mobile/widget/DropdownMenu.js (hunk 1 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/mobile/widget/DropdownMenu.js:1032]: I agree with Grzegorz. This is out of scope this PR.

File: src/js/profile/mobile/widget/DropdownMenu.js [hunk 3]
```diff
@@ -578,30 +587,28 @@
 			};
 
 			/**
-			 * Return offset of element
-			 * @method getOffsetOfElement
+			 * Return top offset of element
+			 * @method getTopOffsetOfElement
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/DropdownMenu.js:591]: You modified description but method name remains old.

File: src/js/profile/mobile/widget/DropdownMenu.js [hunk 4]
```diff
@@ -1073,12 +1086,14 @@
 			 * @protected
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/DropdownMenu.js:1087]: @param {event} [clickEvent=null] click event

**Review by TomaszLukawskiSam — CHANGES_REQUESTED** [review]

File: src/css/profile/mobile/common/dropdownmenu.less (hunk 2 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/common/dropdownmenu.less]: Right but they are still needed.
Those classes (top or bottom) are applied by DropdownMenu.js and are used by css to animate showing menu in specified direction, for example:
```
        &.ui-dropdownmenu-options-opening.ui-dropdownmenu-options-top .ui-dropdownmenu-options {
                .animation(open-to-top 300ms);
        }
```

File: src/js/profile/mobile/widget/DropdownMenu.js (hunk 3 above)
Comment (grzegorz-czajkowski) [reply src/js/profile/mobile/widget/DropdownMenu.js:591]: Done.

File: src/js/profile/mobile/widget/DropdownMenu.js (hunk 4 above)
Comment (grzegorz-czajkowski) [reply src/js/profile/mobile/widget/DropdownMenu.js:1087]: Done.

**Review by TomaszLukawskiSam — APPROVED** [review]
Works good.

**Pull request merged**